src/main/java/redis/clients/jedis/timeseries/TSInfo.java

```diff
@@ -0,0 +1,158 @@
+import redis.clients.jedis.util.SafeEncoder;
+
+public class TSInfo {
+
+  private static final String LABELS_PROPERTY = "labels";
+  private static final String RULES_PROPERTY = "rules";
+  private static final String CHUNKS_PROPERTY = "Chunks";
+  private static final String CHUNKS_BYTES_PER_SAMPLE_PROPERTY = "bytesPerSample";
+
+  private final Map<String, Object> properties;
+  private final Map<String, String> labels;
+  private final Map<String, Rule> rules;
+  private final List<Map<String, Object>> chunks;
+
+  private TSInfo(Map<String, Object> properties, Map<String, String> labels, Map<String, Rule> rules, List<Map<String, Object>> chunks) {
+    this.properties = properties;
+    this.labels = labels;
+    this.rules = rules;
+    this.chunks = chunks;
+  }
+
+  public Map<String, Object> getProperties() {
+    return properties;
+  }
+
+  public Object getProperty(String property) {
+    return properties.get(property);
+  }
+
+  public Long getIntegerProperty(String property) {
+    return (Long) properties.get(property);
+  }
+
+  public Map<String, String> getLabels() {
+    return labels;
+  }
+
+  public String getLabel(String label) {
+    return labels.get(label);
+  }
+
+  public Map<String, Rule> getRules() {
+    return rules;
+  }
+
+  public Rule getRule(String rule) {
+    return rules.get(rule);
+  }
+
+  public List<Map<String, Object>> getChunks() {
+    return chunks;
+  }
+
+  public static Builder<TSInfo> TIMESERIES_INFO = new Builder<TSInfo>() {
+    @Override
+    public TSInfo build(Object data) {
+      List<Object> list = (List<Object>) data;
+      Map<String, Object> properties = new HashMap<>();
+      Map<String, String> labels = null;
+      Map<String, Rule> rules = null;
+      List<Map<String, Object>> chunks = null;
+
+      for (int i = 0; i < list.size(); i += 2) {
+        String prop = SafeEncoder.encode((byte[]) list.get(i));
+        Object value = list.get(i + 1);
+        if (value instanceof List) {
+          switch (prop) {
+            case LABELS_PROPERTY:
+              labels = BuilderFactory.STRING_MAP_FROM_PAIRS.build(value);
+              value = labels;
+              break;
+            case RULES_PROPERTY:
+              List<Object> rulesDataList = (List<Object>) value;
+              List<List<Object>> rulesValueList = new ArrayList<>(rulesDataList.size());
+              rules = new HashMap<>(rulesDataList.size());
+              for (Object ruleData : rulesDataList) {
+                List<Object> encodedRule = (List<Object>) SafeEncoder.encodeObject(ruleData);
+                rulesValueList.add(encodedRule);
+                try {
+                  rules.put((String) encodedRule.get(0),
+                      new Rule((String) encodedRule.get(0), (Long) encodedRule.get(1),
+                          AggregationType.safeValueOf((String) encodedRule.get(2)), (Long) encodedRule.get(3)));
+                } catch (Exception e) { }
```

Comment:
I like to throw exceptions directly here

Reply:
done